### PR TITLE
Fix chai libraries

### PR DESCRIPTION
**Pull request by jednano**
This PR addresses the following issues:
- [x] Missing tests [added](https://github.com/jedmao/DefinitelyTyped/blob/0e5f87c65a05e7224093e7daa49d0b7c67f6369c/chai-fuzzy/chai-fuzzy-tests.ts).
- [x] Indentation and quote type inconsistencies resolved (click [here](https://github.com/borisyankov/DefinitelyTyped/pull/4205/files?w=1) to view this PR w/o whitespace changes).
- ~~[x] [EditorConfig](http://editorconfig.org/) files introduced to assist with preserving indentation settings specific to each folder and not enforced everywhere ([plugin recommended](http://editorconfig.org/#download)).~~
- [x] Fix definitions for Chai and associated Chai libraries (plugins), extending Chai in the proper and consistent way between plugins.
- [x] Resolves any/all compiler errors that were revealed in Visual Studio with ReSharper. There were an insane number of red squigglies all over the place!
- chai
  - [ ] CR
  - [x] Travis-CI
- chai-as-promised
  - [x] CR :+1: 
  - [x] Travis-CI
- chai-datetime
  - [x] CR
  - [x] Travis-CI
- chai-fuzzy
  - [ ] CR
  - [x] Travis-CI
- chai-http
  - [x] CR
  - [x] Travis-CI
- chai-jquery
  - [ ] CR
  - [x] Travis-CI
- sinon
  - [ ] CR
  - [x] Travis-CI
- sinon-chai
  - [ ] CR
  - [x] Travis-CI

### CODE REVIEW

**Comment by jedmao** [comment]
Will the reviewer please review this PR w/o whitespace changes by [clicking here](https://github.com/borisyankov/DefinitelyTyped/pull/4205/files?w=1). Thank you!

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -109,7 +109,7 @@ This document generated by [dt-contributors-generator](https://github.com/vvakam
 * [:link:](camljs/camljs.d.ts) [camljs](http://camljs.codeplex.com) by [Andrey Markeev](http://markeev.com)
 * [:link:](canvasjs/canvasjs.d.ts) [CanvasJS](http://canvasjs.com) by [Mark Overholt](https://github.com/mover5)
 * [:link:](casperjs/casperjs.d.ts) [CasperJS](http://casperjs.org) by [Jed Mao](https://github.com/jedmao)
-* [:link:](chai/chai.d.ts) [chai](http://chaijs.com) by [Jed Hunsaker](https://github.com/jedhunsaker), [Bart van der Schoor](https://github.com/Bartvds)
+* [:link:](chai/chai.d.ts) [chai](http://chaijs.com) by [Jed Mao](https://github.com/jedmao), [Bart van der Schoor](https://github.com/Bartvds)
```
Comment (jedmao) [inline CONTRIBUTORS.md]: I had my name legally changed from Hunsaker to Mao. This reflects that.

**Comment by vvakame** [comment]
* please remove toplevel .editorconfig
* do not edit CONTRIBUTORS.md manually. please fix the definition header. 
* please fix merge conflict.

--- 

*chai-as-promised/chai-as-promised.d.ts*

to author(@jt000). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

---

*chai-datetime/chai-datetime.d.ts*

to author(@cliffburger). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

---

*chai-fuzzy/chai-fuzzy.d.ts*

to author(@Bartvds). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

---

*chai-http/chai-http.d.ts*

to author(@Nemo157). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

---

*chai-jquery/chai-jquery.d.ts*

to author(@kazimanzurrashid). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

---

*chai/chai.d.ts*

to authors(@Bartvds). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

---

*sinon-chai/sinon-chai.d.ts*

to authors(@kazimanzurrashid). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

---

*sinon/sinon.d.ts*

to author(@mrbigdog2u). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

File: chai-as-promised/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,7 @@
+root = true
```
Comment (cliffburger) [inline chai-as-promised/.editorconfig]: There are no other .editorconfig files in the definitely typed repo. Probably best to exclude.

File: chai-fuzzy/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,6 @@
+root = true
```
Comment (cliffburger) [inline chai-fuzzy/.editorconfig]: Again, ```.editorconfig``` should not be include

**Comment by jt000** [comment]
Personally, I like .editorconfig although its only used in a [handful of IDEs](http://editorconfig.org/#download), it does help keep the code consistent. That said, *if* this project were to have an .editorconfig file, then it should only exist once and at the top level of the package. Additionally, it would probably be better to add it in a separate PR so it can get the full attention of everyone.

**Comment by jedmao** [comment]
@jt000 I agree with you that the .editorconfig file should be at the project root only once. Unfortunately, [not everyone agrees](https://github.com/borisyankov/DefinitelyTyped/issues/1569#issuecomment-35524237) and I was requested to remove the file at root; though, I don't see the point, as my root file was only defining the following rules:

```ini
root = true

[*.ts]
trim_trailing_whitespace = true
insert_final_newline = true
```

Without saying anything about indentation, which I put in the folder-specific .editorconfig files with `root = true` removed. Anyway, I'm happy to move the topic of EditorConfig into a separate issue, as it doesn't specifically relate to this PR.

File: chai-as-promised/chai-as-promised.d.ts [hunk 4]
```diff
@@ -32,4 +32,4 @@ declare module chai {
         isRejected(promise: any, expected: any, message?: string): void;
         isRejected(promise: any, match: RegExp, message?: string): void;
     }
-}
\ No newline at end of file
+}
```
Comment (jt000) [inline chai-as-promised/chai-as-promised.d.ts]: Missing trailing }?

Comment (jedmao) [inline chai-as-promised/chai-as-promised.d.ts]: No. That's how it was before. This adds the final newline.

Comment (jt000) [inline chai-as-promised/chai-as-promised.d.ts]: Hmm, my diff in github shows that it was removed... (no newline added). I like the change to Assertion, I think that clears up a lot. :+1:

**Comment by jt000** [comment]
chai-as-promised looks good (just make sure that trailing } wasn't accidentally removed, please) :+1:

File: chai-as-promised/chai-as-promised.d.ts (hunk 4 above)
Comment (jedmao) [inline chai-as-promised/chai-as-promised.d.ts]: Oh, you were talking about the closing curly brace. Yeah. It wasn't removed. GitHub is smoking. See https://github.com/jedmao/DefinitelyTyped/blob/fix-chai-libs/chai-as-promised/chai-as-promised.d.ts

File: chai-as-promised/.editorconfig (hunk 2 above)
Comment (jedmao) [inline chai-as-promised/.editorconfig]: I found numerous issues with inconsistent indentation in these files, even in the same folders, so this aids in preventing those types of issues in the future. I was requested to remove the .editorconfig file at the root, though, which I did.

Comment (jedmao) [inline chai-as-promised/.editorconfig]: There has been more concern about [enforcing a standard project wide](https://github.com/borisyankov/DefinitelyTyped/issues/1569#issuecomment-35524237), but this ensures the rules are only set folder-wide.

File: chai-fuzzy/.editorconfig (hunk 3 above)
Comment (jedmao) [inline chai-fuzzy/.editorconfig]: Do you have a better argument than it not being in other folders? I don't see that as a good reason at all.

Comment (jedmao) [inline chai-fuzzy/.editorconfig]: Is it hurting something in some way? I think not!

File: chai-as-promised/.editorconfig (hunk 2 above)
Comment (jedmao) [inline chai-as-promised/.editorconfig]: @cliffburger I [created an issue](https://github.com/borisyankov/DefinitelyTyped/issues/4211) concerning the EditorConfig files.

**Comment by jedmao** [comment]
@cliffburger I [added a commit](https://github.com/jedmao/DefinitelyTyped/commit/9a7eecbfde6b42579a5cc5ab06b5726951200746) to remove all EditorConfig files until https://github.com/borisyankov/DefinitelyTyped/issues/4211 is addressed. Please give a :+1: if you can!

**Comment by cliffburger** [comment]
chai-datetime looks OK. However, you've updated 19 files. Reviewing the [pull request guidelines](http://definitelytyped.org/guides/pull-request.html). It's possible that all the chai files are interdependent. I'm having another dev here look at it. @akalman

**Comment by cliffburger** [comment]
:+1:

**Comment by Nemo157** [comment]
`chai-http` changes LGTM :+1:

File: chai-datetime/chai-datetime-tests.ts [hunk 5]
```diff
@@ -1,47 +1,50 @@
...
 
 function test_afterDate(){
-    var date: Date = new Date(2014, 1, 1);
-    expect(date).to.afterDate(date);
-    date.should.afterDate(date);
-    assert.afterDate(date, date);
+	var date: Date = new Date(2014, 1, 1);
+	expect(date).to.be.afterDate(date);
+	date.should.be.afterDate(date);
+	assert.afterDate(date, date);
```
Comment (akalman) [inline chai-datetime/chai-datetime-tests.ts]: should we be using spaces or tabs?

File: chai-datetime/chai-datetime.d.ts [hunk 6]
```diff
@@ -5,35 +5,38 @@
...
+		beforeTime(val: Date, exp: Date, msg?: string): void;
+		notBeforeTime(val: Date, exp: Date, msg?: string): void;
+		afterTime(val: Date, exp: Date, msg?: string): void;
+		notAfterTime(val: Date, exp: Date, msg?: string): void;
+		equalDate(val: Date, exp: Date, msg?: string): void;
+		notEqualDate(val: Date, exp: Date, msg?: string): void;
+		beforeDate(val: Date, exp: Date, msg?: string): void;
+		notBeforeDate(val: Date, exp: Date, msg?: string): void;
+		afterDate(val: Date, exp: Date, msg?: string): void;
+		notAfterDate(val: Date, exp: Date, msg?: string): void;
```
Comment (akalman) [inline chai-datetime/chai-datetime.d.ts]: [looking at the implementation](https://github.com/gaslight/chai-datetime/blob/master/chai-datetime.js#L100), I believe these return `Assertion`s.

Comment (akalman) [inline chai-datetime/chai-datetime.d.ts]: nope those are above, derp...

ignore, this is correct.

**Comment by akalman** [comment]
:+1:

File: chai-datetime/chai-datetime-tests.ts (hunk 5 above)
Comment (jedmao) [inline chai-datetime/chai-datetime-tests.ts]: @akalman my mistake. I added a commit to untabify all chai-datetime files.

**Comment by jedmao** [comment]
@cliffburger these chai files are definitely interdependent. If I change chai.d.ts, I need to also change the libs/plugins that reference chai.d.ts. In other words, I can't fix chai.d.ts w/o breaking the libs/plugins that depend on it.

**Comment by jedmao** [comment]
@Bartvds, @kazimanzurrashid, @MrBigDog2U please review this before it becomes stale. Thanks!

**Comment by vvakame** [comment]
I'll review if reviewers do not have reaction in two days.

**Comment by jedmao** [comment]
@vvakame that was 5 days ago. Please review and thank you so much for doing this.

**Comment by vvakame** [comment]
@jedmao it merged.
for reviewers, thank you for reviewing!

**Pull request merged**